pkg/process/v6/transformers/os/transform_test.go

```diff
@@ -1102,6 +1115,128 @@ func TestTransform(t *testing.T) {
 	}
 }
 
+func TestGetOperatingSystem(t *testing.T) {
+	tests := []struct {
+		name      string
+		osName    string
+		osID      string
+		osVersion string
+		expected  *grypeDB.OperatingSystem
+	}{
+		{
+			name:      "works with given args",
+			osName:    "alpine",
+			osID:      "alpine",
+			osVersion: "3.10",
+			expected: &grypeDB.OperatingSystem{
+				Name:         "alpine",
+				ReleaseID:    "alpine",
+				MajorVersion: "3",
+				MinorVersion: "10",
+				LabelVersion: "",
+				Codename:     "",
+			},
+		},
+		{
+			name:      "does codename lookup (debian)",
+			osName:    "debian",
+			osID:      "debian",
+			osVersion: "11",
+			expected: &grypeDB.OperatingSystem{
+				Name:         "debian",
+				ReleaseID:    "debian",
+				MajorVersion: "11",
+				MinorVersion: "",
+				LabelVersion: "",
+				Codename:     "bullseye",
+			},
+		},
+		{
+			name:      "does codename lookup (ubuntu)",
+			osName:    "ubuntu",
+			osID:      "ubuntu",
+			osVersion: "22.04",
+			expected: &grypeDB.OperatingSystem{
+				Name:         "ubuntu",
+				ReleaseID:    "ubuntu",
+				MajorVersion: "22",
+				MinorVersion: "04",
+				LabelVersion: "",
+				Codename:     "jammy",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := getOperatingSystem(tt.osName, tt.osID, tt.osVersion)
+			require.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestGetOSInfo(t *testing.T) {
+	tests := []struct {
+		name       string
+		group      string
+		expectedOS string
+		expectedID string
+		expectedV  string
+	}{
+		{
+			name:       "alpine 3.10",
+			group:      "alpine:3.10",
+			expectedOS: "alpine",
+			expectedID: "alpine",
+			expectedV:  "3.10",
+		},
+		{
+			name:       "debian bullseye",
+			group:      "debian:11",
+			expectedOS: "debian",
+			expectedID: "debian",
+			expectedV:  "11",
+		},
+		{
+			name:       "mariner version 1",
+			group:      "mariner:1.0",
+			expectedOS: "mariner",
+			expectedID: "mariner",
+			expectedV:  "1.0",
+		},
+		{
+			name:       "mariner version 3 (azurelinux conversion)",
+			group:      "mariner:3.0",
+			expectedOS: "azurelinux",
+			expectedID: "azurelinux",
+			expectedV:  "3.0",
+		},
+		{
+			name:       "ubuntu focal",
+			group:      "ubuntu:20.04",
+			expectedOS: "ubuntu",
+			expectedID: "ubuntu",
+			expectedV:  "20.04",
+		},
+		{
+			name:       "oracle linux",
+			group:      "ol:8",
+			expectedOS: "oracle", // normalize name
```

Comment:
Nice - I was having a hard time finding which case would have an OS vs ID difference. This made it very clear